gtp2ogs.js

```diff
@@ -958,44 +936,94 @@ class Game {
                 if (DEBUG) {
                     this.log("State loaded for new bot");
                 }
-            }, passAndRestart);
-        }
+            }, eb);
+        }	
+    } /* }}} */
 
-        if (DEBUG) this.bot.log("Generating move for game", this.game_id);
-        this.log("genmove " + (this.my_color == "black" ? "b" : "w"));
+    // Send @cmd to bot and call @cb with returned moves.
+    //
+    getBotMoves(cmd, cb, eb) { /* {{{ */
+        ++moves_processing;
+        this.processing = true;
+        if (argv.corrqueue && this.state.time_control.speed == "correspondence")
+            ++corr_moves_processing;
 
-        this.bot.genmove(this.state, (move) => {
-            this.processing = false;
+	let doneProcessing = () => {
+            this.procesing = false;
             --moves_processing;
             if (argv.corrqueue && this.state.time_control.speed == "correspondence") {
                 this.corr_move_pending = false;
                 --corr_moves_processing;
             }
-            if (move.resign) {
-                this.log("Resigning");
-                this.socket.emit('game/resign', this.auth({
-                    'game_id': this.state.game_id
-                }));
-            }
-            else {
-                if (DEBUG) this.log("Playing " + move.text, move);
-                else       this.log("Playing " + move.text);
-                this.socket.emit('game/move', this.auth({
-                    'game_id': this.state.game_id,
-                    'move': encodeMove(move)
-                }));
-                //this.sendChat("Test chat message, my move #" + move_number + " is: " + move.text, move_number, "malkovich");
-        if( argv.greeting && !this.greeted && this.state.moves.length < (2 + this.state.handicap) ){
-                  this.sendChat( GREETING, "discussion");
-                  this.greeted = true;
+	};
+
+        let failed = false;
+        let botError = (e) => {
+            if (failed)  return;
+	    
+            failed = true;
+	    doneProcessing();
+            if (this.bot) this.bot.kill();
+            this.bot = null;
+	    if (eb) eb(e);
         }
-            }
-            if (!PERSIST && this.bot != null) {
+
+        if (!this.bot)  this.startBot(botError);
+
+	if (DEBUG) this.bot.log("Generating move for game", this.game_id);
+	this.log("genmove " + (this.my_color == "black" ? "b" : "w"));
```

Comment:
This log message should be updated to use cmd too.

Reply:
Ah, i knew something got clobbered in the rebase =)